### PR TITLE
Allow decimal amounts

### PR DESCRIPTION
**Pull request by Shreyaschorge**
This PR addresses issue https://github.com/Plant-for-the-Planet-org/planet-donations/issues/190

### CODE REVIEW

**Comment by norbertschuler** [comment]
Entering digits works, e.g. for this conservation project https://donate-with-planet-47idl2tvg-planetapp.vercel.app/?to=proj_NPmia9Vh6Ub91nuw2KmWY9uh - but not with German amounts using a comma (",") as a separater which we never really supported (at once place we just automatically replaced every entered comma (",") with a dot (".")).

Bouquet project I could not test as the only active ones on staging do not have custom amounts.

**Comment by tejassonar** [comment]
Maybe we can change the regex for the German language.

File: src/Donations/Micros/DonationTypes/FundingDonations.tsx [hunk 1]
```diff
@@ -205,10 +218,13 @@ function FundingDonations({ setopenCurrencyModal }: Props): ReactElement {
...
-                            /[^0-9]/g,
-                            ""
-                          );
+                          e.target.value = e.target.value
+                            .replace(/[^0-9^\.]+/g, "")
+                            .replace(".", "$#$")
+                            .replace(/\./g, "")
+                            .replace("$#$", ".")
+                            .replace(/^0+/, "");
+
```
Comment (norbertschuler) [inline src/Donations/Micros/DonationTypes/FundingDonations.tsx:227]: So this won't allow to input a "," as used as separater for German values, but afterwards will print the amount, e.g. "5.5." in a localized form e.g. "5,5" for German language. It works and we probably cannot support an input parser for every kind of currency formatting, but I am not sure if everybody will understand this logic.

Comment (norbertschuler) [reply src/Donations/Micros/DonationTypes/FundingDonations.tsx:227]: So close this PR or keep it?

**Comment by mariahosfeld** [comment]
PR is too old, closing this.

**Pull request closed without merging**